src/marks/mark.py

```diff
@@ -97,20 +101,83 @@ def cursorless_mark(m) -> str:
     return m.cursorless_line_number_simple
 
 
-def on_ready():
-    init_csv_and_watch_changes(
-        "special_marks",
+DEFAULT_COLOR_ENABLEMENT = {
+    "blue": True,
+    "green": True,
+    "red": True,
+    "pink": True,
+    "yellow": True,
+}
+
+DEFAULT_SHAPE_ENABLEMENT = {
+    "ex": False,
+    "fox": False,
+    "wing": False,
+    "hole": False,
+    "frame": False,
+    "curve": False,
+    "eye": False,
+    "play": False,
+    "bolt": False,
+    "star": False,
+}
+
+unsubscribe_hat_styles = None
+
+
+def setup_hat_styles_csv():
+    global unsubscribe_hat_styles
+
+    color_enablement = {
+        **DEFAULT_COLOR_ENABLEMENT,
+        **actions.user.vscode_get_setting("cursorless.hatEnablement.colors", {}),
+    }
+    shape_enablement = {
+        **DEFAULT_SHAPE_ENABLEMENT,
+        **actions.user.vscode_get_setting("cursorless.hatEnablement.shapes", {}),
+    }
+
+    active_hat_colors = {
+        spoken_form: value
+        for spoken_form, value in hat_colors.items()
+        if color_enablement[value]
+    }
+    active_hat_shapes = {
+        spoken_form: value
+        for spoken_form, value in hat_shapes.items()
+        if shape_enablement[value]
+    }
+
+    if unsubscribe_hat_styles is not None:
+        unsubscribe_hat_styles()
+
+    unsubscribe_hat_styles = init_csv_and_watch_changes(
+        "hat_styles",
         {
-            "special_mark": special_marks_defaults,
+            "hat_color": active_hat_colors,
+            "hat_shape": active_hat_shapes,
         },
+        [*hat_colors.values(), *hat_shapes.values()],
     )
+
+
+def on_ready():
     init_csv_and_watch_changes(
-        "hat_styles",
+        "special_marks",
         {
-            "hat_color": hat_colors,
-            "hat_shape": hat_shapes,
+            "special_mark": special_marks_defaults,
         },
     )
 
+    setup_hat_styles_csv()
+
+    vscode_settings_path: Path = actions.user.vscode_settings_path()
+
+    def on_watch(path, flags):
+        if vscode_settings_path.match(path):
+            cron.after("500ms", setup_hat_styles_csv)
```

Comment:
Was getting some weird race conditions with `fs.watch` where I'd see the old value sometimes